group-coordinator/src/main/java/org/apache/kafka/coordinator/group/assignor/AssignmentMemberSpec.java

```diff
@@ -29,37 +31,53 @@ public class AssignmentMemberSpec {
     /**
      * The instance ID if provided.
      */
-    final Optional<String> instanceId;
+    private final Optional<String> instanceId;
 
     /**
      * The rack ID if provided.
      */
-    final Optional<String> rackId;
+    private final Optional<String> rackId;
 
     /**
-     * The topics that the member is subscribed to.
+     * The topicIds of topics that the member is subscribed to.
      */
-    final Collection<String> subscribedTopics;
+    private final Collection<Uuid> subscribedTopics;
 
     /**
-     * The current target partitions of the member.
+     * Partitions assigned for this member keyed by topicId
      */
-    final Collection<TopicPartition> targetPartitions;
+    private final Map<Uuid, Set<Integer>> assignedTopicIdPartitions;
```

Comment:
nit: Sorry to repeat myself but how about `assignedPartitions`?

Reply:
I thought it would be more clear if it was topicIdpartitions since we're keying by topicId? Should we change it?